cpp/src/io/orc/writer_impl.cu

```diff
@@ -1472,29 +1474,31 @@ void writer::impl::write(table_view const& table)
   }
 
   // Allocate intermediate output stream buffer
-  size_t compressed_bfr_size   = 0;
-  size_t num_compressed_blocks = 0;
-  auto stream_output           = [&]() {
+  size_t compressed_bfr_size       = 0;
+  size_t num_compressed_blocks     = 0;
+  size_t max_compressed_block_size = 0;
+  if (compression_kind_ != NONE) {
+    nvcompBatchedSnappyCompressGetMaxOutputChunkSize(
```

Comment:
this one is safe to use regardless of the environment variable?

Reply:
Yes. nvcomp is pessimistic about allocation. cuIO is optimistic. So cuIO can always work with that allocation. But the difference is not enough to have any noticeable impact on memory usage or performance.